boa_engine/Cargo.toml

```diff
@@ -50,6 +51,9 @@ unicode-normalization = "0.1.22"
 dyn-clone = "1.0.9"
 once_cell = "1.15.0"
 tap = "1.0.1"
+itertools = "0.10.3"
+sptr = "0.3.2"
+static_assertions = "1.1.0"
```

Comment:
Are these 3 new dependencies really needed? Adding more dependencies increases compile times, and we might not need all of the ones we are using.

Reply:
I'll remove `itertools`, because it introduces a lot of functions we don't really use and it increases compilation times considerably.

`sptr` and `static_assertions` are essential to ensure we're not introducing UB on the `string` module. `sptr` takes care of our pointer provenance, which ensures ptr to int casts won't cause UB in the future. `static_assertions` ensures `COMMON_STRINGS` is not empty, which would cause UB since we're assuming the first string on that array is our default value.

Reply:
`static_assertions` is also a dependency we already use in `boa_interner`.

Reply:
Sounds good :)